### ISSUE TITLE
http://localhost:3000/node_modules/ng2-pagination/ 404 (Not Found)

### ISSUE THREAD

**Issue by Apps-23** [michaelbromley/ng2-pagination#79]
I get this error:

I have tried taking help of this link: https://github.com/michaelbromley/ng2-pagination/issues/44
I have included necessary lines of code as suggested by scottmizo in this link
i.e in my systemjs.config.js
```JavaScript
System.config({ 
  map: {
    'ng2-pagination': 'node_modules/ng2-pagination'
  }, 
  packages: {
   'ng2-pagination ':  { main: 'index.js', defaultExtension: 'js'} 
  }
)
```

in my app.module.ts
import {PaginationControlsCmp, PaginatePipe, PaginationService} from 'ng2-pagination';
```TypeScript
@NgModule({
    imports: [
        BrowserModule,
        FormsModule, 
        routing,
        HttpModule,
        JsonpModule,
        ReactiveFormsModule,
        DndModule.forRoot()
    ],

    exports: [
        BrowserModule,
        DndModule
    ],

    declarations: [
        AppComponent,
        NavBarComponent,
        NavDecorateComponent,
        ContentComponent,
        LoginComponent,
        FavouriteComponent,
        HomeComponent,
        NavBar3Component,
        DefaultProjectComponent,
        DashboardComponent,
        NavBar2Component,
        FilterComponent,
        IssuestatusComponent,
        TrackerComponent,
        FooterComponent,
        PaginationControlsCmp,
        PaginatePipe
    ],

    providers: [

        HTTP_PROVIDERS,
        TokenService,
        ProjectService,
        ValueService,
        PaginationService


    ],



    bootstrap: [
        AppComponent
    ]
})
export class AppModule { }
```

then in the component:
```TypeScript
import {PaginatePipe, PaginationControlsCmp, PaginationService} from 'ng2-pagination';

@Component({
  selector: 'dashboard',
  templateUrl: 'app/templates/dashboard.component.html',
  providers: [CookieService, ProjectService, TokenService, PaginationService],
  directives: [PaginationControlsCmp],
  pipes: [PaginatePipe]
})
```
But error 404 still persists.

**Comment by nacimux** [comment]
I have the same issue !

**Comment by ialex90** [comment]
Use this: 
`import {Ng2PaginationModule} from 'ng2-pagination'; // <-- import the module
`

instead of the components/services:

`import {PaginationControlsCmp, PaginatePipe, PaginationService} from 'ng2-pagination';`

Within the ngModule of the component where you need the pagination.

**Comment by amanullahnxb** [comment]
I have still error even after importing Ng2PaginationModule in my app.module.js.

**Comment by amanullahnxb** [comment]
zone.js:232 Error: (SystemJS) XHR error (404 Not Found) loading http://localhost:3000/node_modules/ng2-pagination(…)(anonymous function) @ localhost/:31ZoneDelegate.invoke @ zone.js:232Zone.run @ zone.js:114(anonymous function) @ zone.js:502ZoneDelegate.invokeTask @ zone.js:265Zone.runTask @ zone.js:154drainMicroTaskQueue @ zone.js:401ZoneTask.invoke @ zone.js:339
http://localhost:3000/traceur Failed to load resource: the server responded with a status of 404 (Not Found)

**Comment by ialex90** [comment]
Hi @amanullahnxb May I ask you to update the comments and refactor them in order to have the error in one, and the code we need to see in the another one? to have the error broken down in 2 pieces and not signal of code anywhere is hard to follow.

It seems to be working, so I would suggest you if you didn't already, read the documentation and the comment above, if still the problem is in there, please do what I just mention about adding info.

**Comment by jspurim** [comment]
There is a typo in the OP config:

```
System.config({ 
  map: {
    'ng2-pagination': 'node_modules/ng2-pagination'
  }, 
  packages: {
   'ng2-pagination ':  { main: 'index.js', defaultExtension: 'js'} 
  }
)
```
should be

```
System.config({ 
  map: {
    'ng2-pagination': 'node_modules/ng2-pagination'
  }, 
  packages: {
   'ng2-pagination':  { main: 'index.js', defaultExtension: 'js'}  //The was an extra space here 
  }
)
```

**Comment by amanullahnxb** [comment]
My error 
`GET http://localhost:3000/node_modules/ng2-pagination/ 404 (Not Found)`

resolved using
//systemjs.config.js
`System.config({ 
  map: {
    'ng2-pagination': 'node_modules/ng2-pagination'
  }, 
  packages: {
   'ng2-pagination':  { main: 'index.js', defaultExtension: 'js'}  //The was an extra space here 
  }
)`



//app.moduls.ts
import {Ng2PaginationModule} from 'ng2-pagination'; // <-- import the module

`imports: [
    ....,
    Ng2PaginationModule,
  .....
  ],`





//template.component.html
`
<ul>
<li *ngFor="let blog of (blogs | paginate: { itemsPerPage: 1, currentPage: p } )">
content ...
</li>
</ul>

<pagination-controls (pageChange)="p = $event"></pagination-controls>
`

**Comment by dennis2gn8** [comment]
I get this error...... GET http://localhost:1337/ng2-pagination 404 (Not Found)
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 var map = {
    'ng2-pagination':             'node_modules/ng2-pagination'
  };
  var packages = {
    'ng2-pagination':             { main: 'index.js',        defaultExtension: 'js'}
  };
HELP!!
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

**Comment by michaelbromley** [comment]
Hi all,

This seems like a problem with the systemjs config. I just took a minute to update the [demo plunker](http://plnkr.co/edit/JVQMPvV8z2brCIzdG3N4?p=preview) to use the latest version of Angular (2.1.1), so please see that for an example of how to configure your systemjs config.

I'm going to close this now, since it is not specifically about this lib, but an external tool.

**Comment by dennis2gn8** [comment]
Thank you very much
